specification/frontdoor/resource-manager/Microsoft.Network/stable/2021-06-09/webapplicationfirewall.json

```diff
@@ -0,0 +1,1038 @@
+          "readOnly": true,
+          "$ref": "#/definitions/ActionType"
+        },
+        "description": {
+          "description": "Describes the functionality of the managed rule.",
+          "readOnly": true,
+          "type": "string"
+        }
+      }
+    },
+    "ManagedRuleExclusion": {
+      "required": [
+        "matchVariable",
+        "selectorMatchOperator",
+        "selector"
+      ],
+      "description": "Exclude variables from managed rule evaluation.",
+      "properties": {
+        "matchVariable": {
+          "type": "string",
+          "enum": [
+            "RequestHeaderNames",
+            "RequestCookieNames",
+            "QueryStringArgNames",
+            "RequestBodyPostArgNames",
+            "RequestBodyJsonArgNames"
+          ],
+          "description": "The variable type to be excluded.",
+          "x-ms-enum": {
+            "name": "ManagedRuleExclusionMatchVariable",
+            "modelAsString": true
+          }
+        },
+        "selectorMatchOperator": {
+          "type": "string",
+          "enum": [
+            "Equals",
+            "Contains",
+            "StartsWith",
+            "EndsWith",
+            "EqualsAny"
+          ],
+          "description": "Comparison operator to apply to the selector when specifying which elements in the collection this exclusion applies to.",
+          "x-ms-enum": {
+            "name": "ManagedRuleExclusionSelectorMatchOperator",
+            "modelAsString": true
+          }
+        },
+        "selector": {
+          "type": "string",
+          "description": "Selector value for which elements in the collection this exclusion applies to."
+        }
+      }
+    },
+    "RuleSetDataBag": {
+      "description": "Holds properties relating to a ruleset and sku association.",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "state": {
+          "$ref": "#/definitions/ManagedRuleSetDefinitionState",
+          "description": "Represents the state of the ruleset."
+        }
+      }
+    },
+    "Features": {
+      "description": "Defines the feature set for a RuleSet.",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "anomalyScoring": {
+          "type": "boolean",
+          "description": "Flag for if AnomalyScoring is supported or not."
+        }
+      }
+    },
+    "ManagedRuleSetDefinitionState": {
+      "description": "Enumeration values for a ManagedRuleSet's state.",
+      "type": "string",
+      "enum": [
+        "GeneralAvailability",
+        "Preview",
+        "Deprecated"
+      ],
+      "x-ms-enum": {
+        "name": "ManagedRuleSetDefinitionState",
+        "modelAsString": true
+      }
+    },
+    "ActionType": {
+      "description": "Defines the action to take on rule match.",
+      "type": "string",
+      "enum": [
+        "Allow",
+        "Block",
+        "Log",
+        "Redirect",
+        "AnomalyScoring"
+      ],
```

Comment:
@disreedh - I don't think this is supported. The API doesn't allow it to be set, it's just the default.

Reply:
CustomRule: throw error InvalidConfigurationException("Custom Rules cannot use Anomaly Scoring action (rule name: {0})", rule.Name);
ManagedRuleOverride: if the rule set version is 2.0 or 2.1 then action can be null or set to Anomaly or Log

Reply:
Spoke offline to both Eric and Divya. Seems like this isn't an issue. Resolving